[BUG] gofiber/fiber v2.50.0 breaking changes

**Version of dd-trace-go**

v1.55.0

**Describe what happened:**

Cannot build a Go project using fiber v2.50.0. This problem arises from the breaking changes in fiber [v2.50.0](https://github.com/gofiber/fiber/releases/tag/v2.50.0). 

```go
- func (c *Ctx) GetReqHeaders() map[string]string
+ func (c *Ctx) GetReqHeaders() map[string][]string
```

**Describe what you expected:**

Project should build without problems.

**Steps to reproduce the issue:**

Install the latest gofiber/fiber and dd-trace-go and setup the middleware like in the example.

```go
// Start the tracer
tracer.Start()
defer tracer.Stop()

// Create a fiber v2 Router
router := fiber.New()

// Use the tracer middleware with the default service name "fiber".
router.Use(fibertrace.Middleware())

// Set up some endpoints.
router.Get("/", func(c *fiber.Ctx) error {
	return c.SendString("test")
})

// And start gathering request traces
router.Listen(":8080")
```

**Additional environment details (Version of Go, Operating System, etc.):**

Build log

```
go build -o ./app ./cmd/server
# gopkg.in/DataDog/dd-trace-go.v1/contrib/gofiber/fiber.v2
../../../go/pkg/mod/gopkg.in/!data!dog/dd-trace-go.v1@v1.55.0/contrib/gofiber/fiber.v2/fiber.go:53:13: cannot use v (variable of type []string) as string value in argument to h.Add
make: *** [Makefile:10: build] Error 1
```

--- a/contrib/gofiber/fiber.v2/fiber.go
+++ b/contrib/gofiber/fiber.v2/fiber.go
@@ -49,8 +49,13 @@ func Middleware(opts ...Option) func(c *fiber.Ctx) error {
 		}
 		// Create a http.Header object so that a parent trace can be extracted. Fiber uses a non-standard header carrier
 		h := http.Header{}
-		for k, v := range c.GetReqHeaders() {
-			h.Add(k, v)
+		for k, headers := range c.GetReqHeaders() {
+			for _, v := range headers {
+				// GetReqHeaders returns a list of headers associated with the given key.
+				// http.Header.Add supports appending multiple values, so the previous
+				// value will not be overwritten.
+				h.Add(k, v)
+			}
 		}
 		if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(h)); err == nil {
 			opts = append(opts, tracer.ChildOf(spanctx))

--- a/go.mod
+++ b/go.mod
@@ -47,7 +47,7 @@ require (
 	github.com/go-redis/redis/v8 v8.11.5
 	github.com/go-sql-driver/mysql v1.6.0
 	github.com/gocql/gocql v0.0.0-20220224095938-0eacd3183625
-	github.com/gofiber/fiber/v2 v2.49.2
+	github.com/gofiber/fiber/v2 v2.50.0
 	github.com/golang/protobuf v1.5.3
 	github.com/gomodule/redigo v1.8.9
 	github.com/google/pprof v0.0.0-20230817174616-7a8ec2ada47b
@@ -79,7 +79,7 @@ require (
 	github.com/tinylib/msgp v1.1.8
 	github.com/twitchtv/twirp v8.1.3+incompatible
 	github.com/urfave/negroni v1.0.0
-	github.com/valyala/fasthttp v1.49.0
+	github.com/valyala/fasthttp v1.50.0
 	github.com/vektah/gqlparser/v2 v2.5.8
 	github.com/zenazn/goji v1.0.1
 	go.mongodb.org/mongo-driver v1.12.1
@@ -115,7 +115,7 @@ require (
 	github.com/DataDog/go-tuf v1.0.2-0.5.2 // indirect
 	github.com/Microsoft/go-winio v0.6.1 // indirect
 	github.com/agnivade/levenshtein v1.1.1 // indirect
-	github.com/andybalholm/brotli v1.0.5 // indirect
+	github.com/andybalholm/brotli v1.0.6 // indirect
 	github.com/armon/go-metrics v0.4.1 // indirect
 	github.com/aws/aws-sdk-go-v2/aws/protocol/eventstream v1.4.13 // indirect
 	github.com/aws/aws-sdk-go-v2/credentials v1.13.20 // indirect
@@ -199,13 +199,13 @@ require (
 	github.com/jmespath/go-jmespath v0.4.0 // indirect
 	github.com/josharian/intern v1.0.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
-	github.com/klauspost/compress v1.16.7 // indirect
+	github.com/klauspost/compress v1.17.1 // indirect
 	github.com/klauspost/cpuid/v2 v2.2.5 // indirect
 	github.com/labstack/gommon v0.4.0 // indirect
 	github.com/leodido/go-urn v1.2.4 // indirect
 	github.com/mailru/easyjson v0.7.7 // indirect
 	github.com/mattn/go-colorable v0.1.13 // indirect
-	github.com/mattn/go-isatty v0.0.19 // indirect
+	github.com/mattn/go-isatty v0.0.20 // indirect
 	github.com/mattn/go-runewidth v0.0.15 // indirect
 	github.com/mitchellh/go-homedir v1.1.0 // indirect
 	github.com/mitchellh/mapstructure v1.5.0 // indirect

